cli/src/cli_util.rs

```diff
@@ -1157,8 +1159,52 @@ impl WorkspaceCommandHelper {
         self.check_repo_rewritable(self.repo().as_ref(), commits)
     }
 
+    fn reset_author_revset(&self) -> Result<Rc<RevsetExpression>, CommandError> {
+        self.reset_author_revset
+            .get_or_try_init(|| {
+                let reset_author = self
+                    .settings
+                    .config()
+                    .get_string("revsets.reset-author-on-edit")
+                    .unwrap_or_default();
+                revset::parse(&reset_author, &self.revset_parse_context()).map_err(|e| {
+                    config_error_with_message("Invalid `revsets.reset-author-on-edit`", e)
+                })
+            })
+            .cloned()
+    }
+
+    fn evaluate_reset_author_revset(
+        mut expression: RevsetExpressionEvaluator,
+        commit_id: &CommitId,
+    ) -> Result<bool, CommandError> {
+        expression.intersect_with(&RevsetExpression::commit(commit_id.clone()));
+        Ok(expression
+            .evaluate_to_commit_ids()
+            .map_err(|e| config_error_with_message("Invalid `revsets.reset-author-on-edit`", e))?
+            .next()
+            .is_some())
+    }
+
+    pub fn check_reset_author(
```

Comment:
nit: perhaps, this can be extracted to non-method function, and `Self::evaluate_reset_author_revset()` can be inlined there.

There aren't many callers, and it's a bit weird that `repo` is passed by argument to `WorkspaceCommandHelper` (which has an associated `repo`.)